packages/cli/src/commands/source/push/delete.ts

```diff
@@ -0,0 +1,164 @@
+      char: 'c',
+      default: true,
+      description:
+        'Specify if children document should also be deleted. Default to `true`.',
+    }),
+  };
+
+  public static args = [
+    {
+      name: 'sourceId',
+      required: true,
+      description:
+        'The identifier of the source on which to perform the delete operation. See source:push:list to obtain the identifier.',
+    },
+  ];
+
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    const {flags} = this.parse(SourcePushDelete);
+    if (!flags.deleteOlderThan && !flags.delete) {
+      this.error(
+        'You must provide either --delete= or --deleteOlderThan=. See source:push:delete --help for more information.'
+      );
+    }
+    const cfg = await new AuthenticatedClient().cfg.get();
+    const source = new Source(cfg.accessToken!, cfg.organization);
+
+    if (flags.deleteOlderThan) {
+      this.doDeletionOlderThan(source);
+      return;
+    }
+
+    if (flags.delete) {
+      if (this.isNumberOfDeletionTooLarge) {
+        this.warn(
+          dedent(
+            'To delete large batch of documents, use source:push:batch command instead.'
+          )
+        );
+        return;
+      }
+
+      await this.doDeletionDocumentURI(source);
+    }
+    await this.config.runHook('analytics', buildAnalyticsSuccessHook(this, {}));
+  }
+
+  private get isNumberOfDeletionTooLarge() {
+    return this.flags.delete.length > 20;
+  }
+
+  private async doDeletionOlderThan(source: Source) {
+    const {flags, args} = this.parse(SourcePushDelete);
+    const toDelete = `older than ${flags.deleteOlderThan}`;
+    try {
+      const res = await source.deleteDocumentsOlderThan(
+        args.sourceId,
+        flags.deleteOlderThan!
+      );
+      this.successMessageOnDeletion(toDelete, res);
+    } catch (e) {
+      this.errorMessageOnDeletion(toDelete, e);
+    }
+  }
+
+  private doDeletionDocumentURI(source: Source) {
+    const {flags, args} = this.parse(SourcePushDelete);
+    return Promise.all(
+      flags.delete.map(async (toDelete) => {
+        try {
+          const res = await source.deleteDocument(
+            args.sourceId,
+            toDelete,
+            flags.deleteChildren
+          );
+          this.successMessageOnDeletion(toDelete, res);
+        } catch (e) {
+          this.errorMessageOnDeletion(toDelete, e);
+        }
+      })
+    );
+  }
+
+  private get flags() {
+    return this.parse(SourcePushDelete).flags;
+  }
+
+  private errorMessageOnDeletion(toDelete: string, e: ErrorFromAPI) {
+    this.warn(
+      dedent(`
+  Error while trying to delete document: ${red(toDelete)}.
+  Status code: ${red(e.response.status)}
+  Error code: ${red(e.response.data.errorCode)}
+  Message: ${red(e.response.data.message)}
+  `)
+    );
+  }
+
+  private successMessageOnDeletion(toDelete: string, res: AxiosResponse) {
+    this.log(
+      dedent(`
+    Successfully deleted document: ${green(toDelete)}
```

Comment:
If you're getting the success code from the Push API, then it doesn't necessarily mean that the operation was successful; it only means that the request was accepted by the Push API. The operation could still fail in the indexing pipeline.

Suggestion:

`The delete request for document: ${green(toDelete)} was accepted by the Push API`

Question: what will `toDelete` contain if only `deleteOlderThan` (not `delete`) was specified?

Reply:
The delete request for document: older than 12345 was accepted by the Push API